### PR TITLE
Implement latest_trade for the Kraken exchange

### PR DESCRIPTION
**Pull request by sdeslauriers**
This commit implements the `latest_trade` method for the Kraken exchange. In addition, it improves error handling of the `_public_query` method.

### CODE REVIEW

File: exchanges/tests/test_kraken.py [hunk 1]
```diff
@@ -34,3 +34,18 @@ def test_server_time(self):
...
+
+    def test_latest_ticker(self):
+        """Test the latest_ticker method"""
+
+        kraken = Kraken()
+
+        # Get the latest trade. I cannot find a way to validate
+        # the result, so I just assume that if it returns something
+        # everything is ok.
+        trade = kraken.latest_trade('XETHZEUR')
```
Comment (gagdiez) [inline exchanges/tests/test_kraken.py]: you know that the result must be of type Trade

File: exchanges/kraken.py [hunk 2]
```diff
@@ -25,15 +30,20 @@ def order_info(self, order_id):
...
+    def _public_query(self, query_name, data=None):
         """Queries public data on Kraken"""
 
         # Request public data. This can raise an exception
         # if there is a connection error or if the request times
         # out. In addition, raise an exception if the status is not
         # OK.
         url = '{}public/{}'.format(self.url, query_name)
-        response = requests.get(url, timeout=self.timeout)
+        response = requests.post(url, data=data, timeout=self.timeout)
```
Comment (gagdiez) [inline exchanges/kraken.py]: Question: Why do we change from get to post? All the public methods are safe, and only used to retrieve data. That is correctly reflected with a "get" request. "Post" should be used only with request that will have some side-effect and need to be secured.

**Review by gagdiez — COMMENTED** [review]
- Not sure about using the post method in public_queries. - test_latest_ticker could be slightly improved.

File: exchanges/kraken.py (hunk 2 above)
Comment (sdeslauriers) [reply exchanges/kraken.py]: The documentation says we should be able to query public methods with GET or POST, but I get an `Invalid parameters` error with GET. 

> "Post" should be used only ...

I disagree, there is no downside to using POST.

Comment (gagdiez) [reply exchanges/kraken.py]: I agree, there's no downside, but the "get" method expresses better what we want to do: to get public data. And I'm not talking only about the action of "getting something", I'm talking about the action of simply asking for something public, with no server side-effect, and which doesn't need any security measure.

Respect to the Invalid parameters error.. requests.get(url, params=data) should work, I can try if you want.

**Review by gagdiez — COMMENTED** [review]
Still not sure about post.

File: exchanges/kraken.py (hunk 2 above)
Comment (sdeslauriers) [reply exchanges/kraken.py]: Your are right, works with the keyword `params`. Strange that get accepts `data`, but does not seem to use it.

**Pull request merged**